### PR TITLE
Git requirements

### LINKED ISSUE #524: Test for presence of git in the relevant commands

### PR DESCRIPTION
**Pull request by ChristianLieven**
* [x] fix #524, #523
* [x] description of feature/fix
* [x] add an entry to the [next release](../HISTORY.rst)

* Include memote's requirement for git on installation instructions (README and docs).
* Check for the presence of `git` in CLI commands that require it.

### CODE REVIEW

File: memote/suite/cli/callbacks.py [hunk 1]
```diff
@@ -97,3 +98,15 @@ def abort_if_false(ctx, param, value):
...
     if not value:
         ctx.abort()
+
+
+def git_installed():
+    """Interrupt execution of memote if `git` has not been installed."""
+    LOGGER.info("Checking `git` installation.")
+    try:
+        check_output(['git', '--version'])
+    except Exception as e:
```
Comment (Midnighter) [inline memote/suite/cli/callbacks.py]:
```suggestion
    except subprocess.CalledProcessError as error:
```

File: memote/suite/cli/callbacks.py [hunk 2]
```diff
@@ -97,3 +98,15 @@ def abort_if_false(ctx, param, value):
...
+        LOGGER.critical(
+            "The execution of memote was interrupted since no installation of "
+            "`git` could be detected. {}".format(str(e)))
```
Comment (Midnighter) [inline memote/suite/cli/callbacks.py]: I think in this case we don't need to attach the exact error message. If you do want to make it possible to show the message I would add below a debug message (see code below). Instead I would add a URL for how to install git, for example, https://git-scm.com/book/en/v2/Getting-Started-Installing-Git

```python
LOGGER.debug("Underlying error:", exc_info=error)
```

File: memote/suite/cli/callbacks.py (hunk 1 above)
Comment (ChristianLieven) [reply memote/suite/cli/callbacks.py]: Done!

File: memote/suite/cli/callbacks.py (hunk 2 above)
Comment (ChristianLieven) [reply memote/suite/cli/callbacks.py]: Done!

**Pull request merged**